ext/mri/bcrypt_ext.c

```diff
@@ -18,7 +18,7 @@ static VALUE bc_salt(VALUE self, VALUE prefix, VALUE count, VALUE input) {
 
     if(!salt) return Qnil;
 
-    str_salt = rb_str_new2(salt);
+    str_salt = rb_str_new(salt, strlen(salt));
```

Comment:
Is this actually necessary to fix the bug?  It seems like `rb_str_new2` [already does that](https://github.com/ruby/ruby/blob/b768dc2207d800e1c462fc4367e89b2ae2f5a919/string.c#L793-L798). (`rb_str_new2` [is just an alias](https://github.com/ruby/ruby/blob/b768dc2207d800e1c462fc4367e89b2ae2f5a919/include/ruby/intern.h#L835) for `rb_str_new_cstr`)